test/scheduler/actuals_test.go

```diff
@@ -371,3 +385,52 @@ func pickFixedPolicySnapshotStatusUpdatedActual(valid, invalidOrNotFound []strin
 		return nil
 	}
 }
+
+func pickAllPolicySnapshotStatusUpdatedActual(scored, filtered []string, policySnapshotName string) func() error {
+	return func() error {
+		policySnapshot := &placementv1beta1.ClusterSchedulingPolicySnapshot{}
+		if err := hubClient.Get(ctx, types.NamespacedName{Name: policySnapshotName}, policySnapshot); err != nil {
+			return err
+		}
+
```

Comment:
compare the full struct of the policy status?

`If your function returns a struct, don't write test code that performs an individual comparison for each field of the struct. Instead, construct the struct that you're expecting your function to return, and compare in one shot using [diffs](https://github.com/golang/go/wiki/TestComments#print-diffs) or [deep comparisons](https://github.com/golang/go/wiki/TestComments#equality-comparison-and-diffs). The same rule applies to arrays and maps.
`
https://github.com/golang/go/wiki/TestComments#compare-full-structures

Reply:
Again and again, any Go guideline is just a general guideline, not a must. I don't particularly have a strong opinion on this one, but the check here is done separately to a) avoid having a large bloc of data and b) avoid stuffing the Diff() call with tons of options, which is even more difficult to reason about.

Reply:
you can always construct the options using,

snapshotStatusCmpOptions = cmp.Options{
		ignoreClusterDecisionReasonField,
                cmpopts.SortSlices(lessFuncClusterDecision),
               ...
	}

and build the wantStatus like

wantStatus := SchedulingPolicySnapshotStatus{
         	ObservedCRPGeneration: wantCRPGeneration
                Conditions: []metav1.Condition{
                          wantScheduledCondition,
                }
               ClusterDecisions: wantClusterDecisions,
}

Then compare the status in one Diff call, the code will be cleaner than comparing the field one by one.
The amount of the data is the same as the current one.

Reply:
Stuffing all the options into one struct/array does not make them disappear. And building this want struct with blocks IMO does not help much with readability.